zuul-core/src/main/java/com/netflix/zuul/netty/server/psk/TlsPskHandler.java

```diff
@@ -0,0 +1,499 @@
+            return switch (cipherSuite) {
+                case CipherSuite.TLS_AES_128_CCM_SHA256,
+                        CipherSuite.TLS_AES_128_CCM_8_SHA256,
+                        CipherSuite.TLS_AES_128_GCM_SHA256,
+                        CipherSuite.TLS_CHACHA20_POLY1305_SHA256 -> PRFAlgorithm.tls13_hkdf_sha256;
+                case CipherSuite.TLS_AES_256_GCM_SHA384 -> PRFAlgorithm.tls13_hkdf_sha384;
+                case CipherSuite.TLS_SM4_CCM_SM3, CipherSuite.TLS_SM4_GCM_SM3 -> PRFAlgorithm.tls13_hkdf_sm3;
+                default -> -1;
+            };
+        }
+    }
+
+    static class TlsPskServerProtocol extends TlsServerProtocol {
+
+        public SSLSession getSSLSession() {
+            return new SSLSession() {
+                @Override
+                public byte[] getId() {
+                    return tlsSession.getSessionID();
+                }
+
+                @Override
+                public SSLSessionContext getSessionContext() {
+                    return null;
+                }
+
+                @Override
+                public long getCreationTime() {
+                    return 0;
+                }
+
+                @Override
+                public long getLastAccessedTime() {
+                    return 0;
+                }
+
+                @Override
+                public void invalidate() {}
+
+                @Override
+                public boolean isValid() {
+                    return !isClosed();
+                }
+
+                @Override
+                public void putValue(String name, Object value) {}
+
+                @Override
+                public Object getValue(String name) {
+                    return null;
+                }
+
+                @Override
+                public void removeValue(String name) {}
+
+                @Override
+                public String[] getValueNames() {
+                    return new String[0];
+                }
+
+                @Override
+                public Certificate[] getPeerCertificates() throws SSLPeerUnverifiedException {
+                    return new Certificate[0];
+                }
+
+                @Override
+                public Certificate[] getLocalCertificates() {
+                    return new Certificate[0];
+                }
+
+                @Override
+                public X509Certificate[] getPeerCertificateChain() throws SSLPeerUnverifiedException {
+                    return new X509Certificate[0];
+                }
+
+                @Override
+                public Principal getPeerPrincipal() throws SSLPeerUnverifiedException {
+                    return null;
+                }
+
+                @Override
+                public Principal getLocalPrincipal() {
+                    return null;
+                }
+
+                @Override
+                public String getCipherSuite() {
+                    return SUPPORTED_TLS_PSK_CIPHER_SUITE_MAP.get(
```

Comment:
Is the suite map the accurate value here? 
Or should it be derived from the underlying session, i.e. `SessionParameters` in `this.getContext().getSession()`?

Reply:
I'm actually getting the negotiated cipher suite from context `getContext().getSecurityParameters().getCipherSuite()` which returns an integer but `SSLSession` requires me to return a name of the suite, the map is actually doing the conversion from int to String.